### PR TITLE
Support for a few timestamp casts

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>

### CODE REVIEW

File: lib/sqllib/src/casts.rs [hunk 1]
```diff
@@ -1985,3 +2000,15 @@ pub fn cast_to_ShortInterval_i64(value: i64) -> ShortInterval
 {
     ShortInterval::from(value)
 }
+
+//////// casts to Timestamp
+
+#[inline]
+pub fn cast_to_Timestamp_s(value: String) -> Timestamp
+{
+    let r = NaiveDateTime::parse_from_str(&value, "%Y %m %d %H:%M:%S");
```
Comment (lalithsuresh) [inline lib/sqllib/src/casts.rs:2009]: Is there any guarantee that value is represented with this format string "%Y %m %d %H:%M:%S" ?

Comment (mbudiu-vmw) [reply lib/sqllib/src/casts.rs:2009]: no, it comes from a string.
at least some databases will parse this form correctly

Comment (lalithsuresh) [reply lib/sqllib/src/casts.rs:2009]: No, I mean, doesn't this code assume that the cast_to_Timestamp_s() caller always passes a string formatted according to the formatting assumption above? Would it work with an input string "12:54:10pm 01/02/2010" ?

Comment (mbudiu-vmw) [reply lib/sqllib/src/casts.rs:2009]: Probably not. Should it?

Comment (lalithsuresh) [reply lib/sqllib/src/casts.rs:2009]: Depends on what queries we want to support, and I'd image queries involving dates might routinely specify their format strings.

Comment (mbudiu-vmw) [reply lib/sqllib/src/casts.rs:2009]: I got this format from some sql server spec.
There are much more precise functions, such as convert, which can handle formats.
cast is just cast.
I am happy if it has the property that going back and forth is the identity function.

Comment (lalithsuresh) [reply lib/sqllib/src/casts.rs:2009]: Okay, sounds good. We can leave a todo there to test against more formats and merge for now.

**Pull request merged**